### PR TITLE
[swappable] Further improvements #104 #108

### LINKED ISSUE #104: [swappable] Further improvements after #102
Address the comments and suggestions in the following review:
https://github.com/openwisp/openwisp-monitoring/pull/102#pullrequestreview-420398266

### PR DESCRIPTION
**Pull request by nepython**
Closes #104 #108

I read the docs of celery and found that `celery` auto detects `tasks` of only `INSTALLED_APPS`. Hence, `openwisp_monitoring.device.tasks` was not getting auto-detected but I have explicitly declared it and now it's a registered task with CELERY. I think `openwisp_monitoring.check.tasks` was detected since `run_checks` task was declared for explicitly in `settings` for `celery beat` (I am unsure)

I also changed the test `test_last_called` in `sample_check` to take care of edge cases.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -795,7 +795,25 @@ file in the test project.
...
 
-13. Import the automated tests
+13. Import Celery Tasks
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Add the following in your settings.py to import celery tasks from ``device_monitoring`` app.
+
+.. code-block:: python
+
+    CELERY_IMPORTS = ('openwisp_monitoring.device.tasks')
```
Comment (PabloCastellano) [inline README.rst:805]: This is a string and a tuple is expected. Either you add a trailing comma or convert it to a list

File: tests/openwisp2/sample_check/tests.py [hunk 2]
```diff
@@ -29,10 +29,9 @@ def test_last_called(self):
             params={'count': 2, 'interval': 10, 'bytes': 10, 'timeout': 50},
         )
         check.perform_check(store=False)
-        # Left the seconds due to minor time lag
-        self.assertEqual(
-            check.last_called.strftime('%D %H %M'), now().strftime('%D %H %M')
-        )
+        # Avoid any failures due to minor lag
+        lag = int(now().strftime('%s')) - int(check.last_called.strftime('%s'))
+        assert lag < 4
```
Comment (PabloCastellano) [inline tests/openwisp2/sample_check/tests.py:34]: There exist `self.assertLess` and `self.assertLessEqual`

File: tests/openwisp2/sample_monitoring/tests.py [hunk 3]
```diff
@@ -17,7 +17,20 @@ class TestDatabase(BaseTestDatabase):
 
 
 class TestModels(BaseTestModels):
-    pass
+    def test_metric_details_field(self):
```
Comment (PabloCastellano) [inline tests/openwisp2/sample_monitoring/tests.py:20]: Why is this test specific for sample_monitoring?

File: README.rst (hunk 1 above)
Comment (nepython) [reply README.rst:805]: I had added a `tuple` not sure if it occured due to `black`.

File: tests/openwisp2/sample_monitoring/tests.py (hunk 3 above)
Comment (nepython) [reply tests/openwisp2/sample_monitoring/tests.py]: I didn't get the query. I have added some additional fields for every model as an example for the user and @nemesisdesign suggested to have a test for testing the same. (You can check the suggestions in #102). Besides that, I am unsure if you implied something else.

Comment (PabloCastellano) [reply tests/openwisp2/sample_monitoring/tests.py]: Fair enough, it just seemed odd that this is just the only test in the sample_app

File: tests/openwisp2/settings.py [hunk 4]
```diff
@@ -167,6 +167,9 @@
     },
 }
 
+# Required for SAMPLE_APP as celery auto detects tasks only from INSTALLED_APPS
+CELERY_IMPORTS = ('openwisp_monitoring.device.tasks',)
```
Comment (nemesisdesign) [inline tests/openwisp2/settings.py:171]: move this to the `if os.environ.get('SAMPLE_APP', False):` block, otherwise we won't spot problems if we mess the  autoloading of celery tasks

**Comment by nepython** [comment]
>Missing:
>*  #102 (comment)

@nemesisdesign, I actually didn't get the need for it. If I understood it correctly from the docs, we can supply the `test` management command with **(1)** the names of the apps that need to be run one after the other as arguements or **(2)** the root directory containing all the apps will also suffice. In the non-sample apps case we are doing the **(2)** while in the sample-apps case `notifications` app's tests were not getting executed so I explicitly mentioned it **(1)**.

If I am missing or have misunderstood anything, please let me know.
